TikTokTrainer/UI/VideoPreview.swift

```diff
@@ -53,12 +76,41 @@ class QueuePlayerUIView: UIView {
     }
 }
 
+class PlayerUIView: UIView {
+    private var playerLayer = AVPlayerLayer()
+    private var player: AVPlayer?
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        let playerItem = AVPlayerItem(url: videoURL)
+
+        player = AVPlayer(playerItem: playerItem)
+        playerLayer.player = player
+        layer.addSublayer(playerLayer)
+
+        player?.play()
+    }
+
+    func playVideo() {
+        player?.play()
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        playerLayer.frame = bounds
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
 struct ProgressBar: UIViewRepresentable {
     var duration: Double = 0.0
     var progressBar = UIProgressView()
 
     func makeUIView(context: Context) -> some UIView {
-        print(duration)
         DispatchQueue.main.async {
             UIView.animate(withDuration: self.duration) {
                 progressBar.setProgress(1.0, animated: true)
```

Comment:
We are using progress view here?

Reply:
No this is a UIKit UIProgressView while a ProgressView is from SwiftUI and you can just pass the state duration and it will handle the rest. Like I think this entire struct can be removed and replaced with a two line ProgressView call in the main view

Reply:
Is there a good reason for changing it? The functionality is the same except it seems the standard for animating swiftui progress bar is with a timer which is choppy unless we call the timer function every .05 seconds.